sql/connect/common/src/main/scala/org/apache/spark/sql/connect/client/arrow/ArrowDeserializer.scala

```diff
@@ -336,6 +337,34 @@ object ArrowDeserializers {
           }
         }
 
+      case (VariantEncoder, StructVectors(struct, vectors)) =>
```

Comment:
In the future I would like to move to a situation where a client actually gets an object they can interact with. So instead of two arrays, actually produce a Row with schema (or a Map).